app/assets/javascripts/pretty-text/engines/discourse-markdown/upload-protocol.js

```diff
@@ -6,8 +6,9 @@ const HTML_TYPES = ["html_block", "html_inline"];
 function addImage(uploads, token) {
   if (token.attrs) {
     for (let i = 0; i < token.attrs.length; i++) {
-      if (token.attrs[i][1].indexOf("upload://") === 0) {
-        uploads.push({ token, srcIndex: i, origSrc: token.attrs[i][1] });
+      const value = token.attrs[i][1];
+      if (value?.indexOf("upload://") === 0) {
```

Comment:
It's always a good time to start using nice functions! 😁 

```suggestion
      if (value?.startsWith("upload://")) {
```